<launch>

  <arg name="robot" default="0"/>

  <group if="$(eval arg('robot') == 0)">
    <machine name="turtlebot" address="localhost"/>
  </group>


  <group if="$(eval arg('robot') != 0)">
    <machine name="turtlebot" address="turtlebot1" env-loader="/home/student/luxi_install/env.sh" user="student"  />
  </group>

  <node name="serial_node" pkg="rosserial_python" type="serial_node.py" machine="turtlebot">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="115200"/>
  </node>



  <node name="turtle_interface" pkg="nuturtle_robot" type="turtle_interface" output="screen">
  </node>

  <node name="trans" pkg="nuturtle_robot" type="trans" output="screen">
  </node>

  <node name="fake_odometer" pkg="rigid2d" type="odometer" output="screen">
    <param name="joint_state_topic" value="/fake/joint_states"/>
    <param name="odom_frame_id" value="/fake_odom_frame_id"/>
    <param name="base_link" value="/fake_base_link"/>
    <remap from="odom" to="fake_odom"/>
    <remap from="joint_state" to="/fake/joint_states"/>
  </node>

  <node name="fake_diff_encoders" pkg="rigid2d" type="fake_diff_encoders" output="screen">
    <param name="joint_state_topic" value="/fake/joint_states"/>
  </node>

  <node name="odometer" pkg="rigid2d" type="odometer" output="screen">
    <param name="joint_state_topic" value="/joint_states"/>
    <param name="odom_frame_id" value="/odom_frame_id"/>
    <param name="base_link" value="/base_link"/>
  </node>

  <param name="robot_description"
       command="$(find xacro)/xacro $(find nuturtle_description)/urdf/diff_drive.urdf.xacro"/>
  <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" /> -->


  <node pkg="tf2_ros" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 odom_frame_id fake_odom_frame_id" />

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find nuturtle_robot)/config/config.rviz" />

</launch>
